mlrun/model_monitoring/db/tsdb/tdengine/tdengine_connector.py

```diff
@@ -260,13 +325,65 @@ def read_predictions(
         start: datetime,
         end: datetime,
         aggregation_window: typing.Optional[str] = None,
+        limit: typing.Optional[int] = None,
     ) -> typing.Union[
         mm_schemas.ModelEndpointMonitoringMetricValues,
         mm_schemas.ModelEndpointMonitoringMetricNoData,
     ]:
-        raise NotImplementedError
+        df = self.get_records(
+            table=mm_schemas.TDEngineSuperTables.PREDICTIONS,
+            start=start,
+            end=end,
+            columns=["latency"],
+            filter_query=f"endpoint_id='{endpoint_id}'",
+            agg_funcs=["count"],
+            interval=aggregation_window,
+            limit=limit,
+        )
+
+        full_name = mlrun.model_monitoring.helpers.get_invocations_fqn(self.project)
+
+        if df.empty:
+            return mm_schemas.ModelEndpointMonitoringMetricNoData(
+                full_name=full_name,
+                type=mm_schemas.ModelEndpointMonitoringMetricType.METRIC,
+            )
+
+        if not aggregation_window:
```

Comment:
I still don't think it's consistent with TSDB... Maybe just make it mandatory?

Reply:
it's not mandatory with V3IO so that's why I kept the same behaviour here as well. I think it's OK because in that case when you call `read_predictions()` you get just a single value which is the total amount of predictions in the provided time range.